packages/react-components/src/components/SendBoxErrorBar.tsx

```diff
@@ -23,15 +23,19 @@ export interface SendBoxErrorBarProps {
 export const SendBoxErrorBar = (props: { message?: string; timeout?: number }): JSX.Element => {
   const { message, timeout } = props;
   const [errorMessage, setErrorMessage] = React.useState(message);
+  const setTimeoutRef = React.useRef<NodeJS.Timeout>();
 
   useEffect(() => {
+    const clearSetTimeout = (): void => {
+      setTimeoutRef.current && clearTimeout(setTimeoutRef.current);
+    };
+
+    clearSetTimeout();
     if (timeout !== undefined) {
-      const messageTimeout = setTimeout(() => {
+      setTimeoutRef.current = setTimeout(() => {
         setErrorMessage(undefined);
       }, timeout);
-      return () => {
-        clearTimeout(messageTimeout);
-      };
+      return clearSetTimeout;
     } else {
       return;
```

Comment:
if else is just a return do we need to even write line 39-41?

Reply:
This is for preventing warning:
Not all code paths return

Reply:
But it's just a warning so I am gonna remove it.